device/pci_device.cpp

```diff
@@ -0,0 +1,431 @@
+/*
+ * SPDX-FileCopyrightText: (c) 2023 Tenstorrent Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <cstdint>
+#include <vector>
+#include <fcntl.h>  // for ::open
+#include <unistd.h> // for ::close
+#include <sys/ioctl.h> // for ioctl
+#include <sys/mman.h>  // for mmap, munmap
+#include <linux/pci.h> // for PCI_SLOT, PCI_FUNC
+
+#include "pci_device.hpp"
+#include "architecture_implementation.h"
+#include "ioctl.h"
+#include "device/tt_arch_types.h"
+#include "device/driver_atomics.h"
+
+#include "common/assert.hpp"
+#include "common/logger.hpp"
+
+int find_device(const uint16_t device_id) {
+    // returns device id if found, otherwise -1
+    const char device_name_pattern [] = "/dev/tenstorrent/%u";
+    char device_name[sizeof(device_name_pattern) + std::numeric_limits<unsigned int>::digits10];
+    std::snprintf(device_name, sizeof(device_name), device_name_pattern, (unsigned int)device_id);
+    int device_fd = ::open(device_name, O_RDWR | O_CLOEXEC);
+    // LOG2 ("find_device() open call returns device_fd: %d for device_name: %s (device_id: %d)\n", device_fd, device_name, device_id);
+    return device_fd;
+}
+
+tt::ARCH detect_arch(uint16_t pcie_device_id, int pcie_revision_id) {
+    if (pcie_device_id == 0xfaca){
```

Comment:
Maybe we could put `0xfaca` and other pcie device ids into some define or constexpr variables. I think it would be more readable and since we are already using `0x401e` twice below it makes errors much more possible when you need to retype pcie device ids

Reply:
These were previously in functions, which have now been deleted. I agree that constants for these would be more readable.

On the other hand, I'd be fine if this is not addressed in this PR if it burdens it, but rather if an issue is created to get back to this and also look at other constants throughout the code if needed.

Reply:
addressed in new PR